### DISCUSSION TITLE
Data Format for Regression

### DISCUSSION THREAD

**Discussion by asarm** [alan-turing-institute/sktime#4026] [Q&A]
I am trying to use any regressor model from sktime but I couldn't figure out how to create the data type and format I need to use. Assume I want to use 2 columns as input and 1 column as target.

```
from sktime.regression.interval_based import TimeSeriesForestRegressor

rand = np.random.random((200, 3))
X = pd.DataFrame(rand)[[0, 1]]
y = pd.DataFrame(rand)[2]

forecaster = TimeSeriesForestRegressor()
forecaster.fit(X=X, y=pd.Series(y))
```

Code block above this error: "X is not of a supported input data type.X must be in a supported mtype format for Panel, found <class 'pandas.core.frame.DataFrame'>Use datatypes.check_is_mtype to check conformance with specifications."

How do I convert my data to a format suitable for sktime as a time series task?

**Comment by fkiraly** [comment]
I would recommend you take a look at the classification tutorial,
https://mybinder.org/v2/gh/sktime/sktime/main?filepath=examples

the format requirements (and options) are exactly the same as for regression.

**Reply by asarm** [reply]
Yes, I got the solution. Thanks.
